build.gradle

```diff
@@ -35,7 +39,28 @@ java {
 }
 
 tasks.withType(JavaCompile).configureEach {
+    // prevent compiling MC with newer java versions
+    if (it.name in ['compileMcLauncherJava', 'compilePatchedMcJava']) {
+        return
+    }
```

Comment:
Why return here, and not just stick it with `compileTestJava` so it is run with java 8?